client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -554,6 +540,14 @@ boolean contentSubscribed(ResponseContentFlowable content) {
             }
         }
 
+        private void releaseAndCloseIfRequestNotComplete() {
+            if (finishedWritingRequestBody) {
+                channelPool.release(channel);
+            } else {
+                closeAndReleaseChannel();
```

Comment:
There might be a race here where we close a channel that didn't necessarily need to be closed. finishedWritingRequestBody is assigned by one thread between this thread executing lines 544 and 547.